### ISSUE TITLE
autoload typo

### ISSUE THREAD

**Issue by return42** [Lindydancer/ini-mode#1]
Thanks for this smart ini-mode, but there might be a typo with the _autoload cookie_ (`;;;autoload`).
- https://www.gnu.org/software/emacs/manual/html_node/elisp/Autoload.html

You added the call `ini-define-prog-mode ini-mode` to the _autoloads_, while the `ini-define-prog-mode ini-mode` macro is defined in the module you want to _autoload_ ;-)

Here is my suggested patch ...

``` diff
diff --git a/common/site-lisp/ini-mode.el b/common/site-lisp/ini-mode.el
index e812996..c149042 100644
--- a/common/site-lisp/ini-mode.el
+++ b/common/site-lisp/ini-mode.el
@@ -82,11 +82,11 @@ If `prog-mode' is defined, inherit from it."
      ,mode ,(and (fboundp 'prog-mode) 'prog-mode)
      ,name ,@args))

-;;;###autoload
 (ini-define-prog-mode ini-mode "ini"
   "Major mode for editing Windows-style ini files."
   (setq font-lock-defaults '(ini-font-lock-keywords nil)))

+;;;###autoload(autoload 'ini-mode "ini-mode" nil t)
 ;;;###autoload(add-to-list 'auto-mode-alist '("\\.ini\\'" . ini-mode))

 ;;; ini-mode.el ends here
```

**Comment by Lindydancer** [comment]
Hi!

First, I'm really sorry I haven't responded on this until now. Things have
been very hectic for a while, and the little Emacs time I had I've spent
working on the upcoming Emacs 25 release.

Yes, you are right, the autoload:s doesn't work in the current version. I
did notice this a while ago and I've been experimenting with different
solutions. I also dived into the source code of the autoload cookie parser
and realised that it is aware of a handful of special cases, like
`define-major-mode', which mean that it won't work with locally defined
macros.

Each method I've tested have had it's advantages and drawbacks. One
drawback with the solution you suggested is that the doc string isn't
included in the autoload. Another is that it's not possible to generate
autoloads with path information -- as it hard-codes "init-mode", it must be
in the path. Despite this, I think that it's the way I intend to go with
this, and other, major modes I've written.

Thanks for reporting this!

Since I'm using "svn" internally and "git svn" to publish my packages, I
won't be able to accept the pull require in a normal GitHub manner as "git
svn" is a one-way system.

```
-- Anders
```
